internal/keepers/ocr.go

```diff
@@ -86,19 +86,23 @@ func (k *keepers) Report(ctx context.Context, rt types.ReportTimestamp, _ types.
 		return false, nil, fmt.Errorf("%w: failed to sort/dedupe attributed observations: %s", err, lCtx)
 	}
 
+	// TODO: Generate multiple reports
 	// select, verify, and build report
 	toPerform := make([]ktypes.UpkeepResult, 0, 1)
 	for _, key := range keys {
-
-		upkeep, err := k.service.CheckUpkeep(ctx, key)
+		upkeeps, err := k.service.CheckUpkeep(ctx, key)
 		if err != nil {
 			return false, nil, fmt.Errorf("%w: failed to check upkeep from attributed observation: %s", err, lCtx)
 		}
 
-		if upkeep.State == ktypes.Eligible {
+		if len(upkeeps) != 1 {
```

Comment:
I see a potential silent failure here. Expected cases are len == 0 || len == 1. If len > 1 we should return an error because that would be unexpected state.